app/controllers/concerns/curation_concerns/curation_concern_controller.rb

```diff
@@ -201,7 +201,19 @@ def curation_concern_from_search_results
       def search_result_document(search_params)
         _, document_list = search_results(search_params)
         raise CanCan::AccessDenied.new(nil, :show) if document_list.empty?
-        document_list.first
+        document = document_list.first
+        if user_cannot_access_due_to_workflow?(document)
+          flash[:notice] = I18n.t("curation_concerns.workflow.default.unauthorized")
```

Comment:
It's using `default` workflow here, even though this may not be the default workflow. If we aren't going to use the workflow name, I'd go with `curation_concerns.workflow.unauthorized`